### PR TITLE
Login functionality

### PR DESCRIPTION
**Pull request by parw8649**

### CODE REVIEW

File: Android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -29,6 +28,24 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity
```
Comment (bKondo) [inline Android/app/src/main/AndroidManifest.xml:31]: We should only have 2 activities, MainAppActivity and LoginAndSignUpActivity. The rest are fragments connected by the navigation graphs.

File: Android/app/src/main/java/com/example/boardgamesocial/DataClasses/User.java [hunk 2]
```diff
@@ -58,11 +56,21 @@ public User(Date dateJoined, String email, String firstName, Boolean isActive, B
...
     }
 
+    //Used for Login
     public User(String username, String password) {
         this.username = username;
         this.password = password;
     }
 
+    //Used for signUp
+    public User(String firstName, String lastName, String email, String username,  String password) {
```
Comment (bKondo) [inline Android/app/src/main/java/com/example/boardgamesocial/DataClasses/User.java:66]: why aren't we using the 2nd constructor to fill out all user fields?

File: Android/app/src/main/java/com/example/boardgamesocial/LoginFragment.java [hunk 3]
```diff
@@ -52,4 +64,40 @@ public void onDestroyView() {
...
+        retrofitClient.loginCall(new User(username, password)
+        ).enqueue(new Callback<Token>() {
+            @Override
+            public void onResponse(Call<Token> call, Response<Token> response) {
+                if (response.isSuccessful()) {
+                    assert response.body() != null;
+                    token = response.body();
+                    Log.i("Token", token.toString());
+                    Toast.makeText(getContext(), "Login Successful", Toast.LENGTH_LONG).show();
+                    Intent goToHomePostActivity = HomePostActivity.getIntent(getContext());
```
Comment (bKondo) [inline Android/app/src/main/java/com/example/boardgamesocial/LoginFragment.java:84]: remember that we changed to a two activity setup
```suggestion
                    Intent goToHomePostActivity = MainAppActivity.getIntent(getActivity().getApplicationContext());
                    startActivity(goToHomePostActivity);
```

**Review by BorisMarin8004 — APPROVED** [review]
Nice!

**Review by bKondo — APPROVED** [review]

**Review by keyoni — APPROVED** [review]

**Pull request merged**